### PR TITLE
[postgres] Add tests for latest offset strategy

### PR DESCRIPTION
**Pull request by loserwang1024**
### What is the purpose of the change
Add tests for latest offset strategy.

### CODE REVIEW

**Comment by loserwang1024** [comment]
@ruanhang1993 , CC, WDYT?

**Comment by loserwang1024** [comment]
@ruanhang1993 , CC,

File: flink-cdc-connect/flink-cdc-source-connectors/flink-connector-postgres-cdc/src/test/java/com/ververica/cdc/connectors/postgres/source/PostgresSourceITCase.java [hunk 1]
```diff
@@ -178,29 +192,39 @@ public void testJobManagerFailoverSingleParallelism() throws Exception {
...
+                assertTrue(
+                        ExceptionUtils.findThrowableWithMessage(
+                                        e,
+                                        String.format(
+                                                "Incremental snapshot for tables requires primary key, but table %s doesn't have primary key",
+                                                SCHEMA_NAME + ".customers_no_pk"))
+                                .isPresent());
+            }
+        } else {
             testPostgresParallelSource(
```
Comment (ruanhang1993) [inline flink-cdc-connect/flink-cdc-source-connectors/flink-connector-postgres-cdc/src/test/java/com/ververica/cdc/connectors/postgres/source/PostgresSourceITCase.java:215]: Other startup mode will not throw this exception?

File: flink-cdc-connect/flink-cdc-source-connectors/flink-connector-postgres-cdc/src/test/java/com/ververica/cdc/connectors/postgres/source/PostgresSourceITCase.java [hunk 2]
```diff
@@ -254,6 +278,8 @@ public void testDebeziumSlotDropOnStop() throws Exception {
                 tableResult, FailoverType.JM, FailoverPhase.STREAM, new String[] {"customers"});
 
         tableResult.getJobClient().get().cancel().get();
+        Thread.sleep(1000L);
+        customDatabase.removeSlot(slotName);
```
Comment (ruanhang1993) [inline flink-cdc-connect/flink-cdc-source-connectors/flink-connector-postgres-cdc/src/test/java/com/ververica/cdc/connectors/postgres/source/PostgresSourceITCase.java:282]: It is better to use `try ... finally ...`.

File: flink-cdc-connect/flink-cdc-source-connectors/flink-connector-postgres-cdc/src/test/java/com/ververica/cdc/connectors/postgres/source/PostgresSourceITCase.java (hunk 1 above)
Comment (loserwang1024) [reply flink-cdc-connect/flink-cdc-source-connectors/flink-connector-postgres-cdc/src/test/java/com/ververica/cdc/connectors/postgres/source/PostgresSourceITCase.java:215]: @ruanhang1993 , this exception is thrown when use primary keys to split table chunks. Other startup modes of Postgres cdc connector no need to generate snapshot split, thus no exception is thrown.
By the way, Postgres cdc connector now only has two startup modes: "initial" and "latest-offset". Thus, I just add test for  latest-offset  strategy.

File: flink-cdc-connect/flink-cdc-source-connectors/flink-connector-postgres-cdc/src/test/java/com/ververica/cdc/connectors/postgres/source/PostgresSourceITCase.java (hunk 2 above)
Comment (loserwang1024) [reply flink-cdc-connect/flink-cdc-source-connectors/flink-connector-postgres-cdc/src/test/java/com/ververica/cdc/connectors/postgres/source/PostgresSourceITCase.java]: Add it now.

**Review by ruanhang1993 — APPROVED** [review]
LGTM

**Pull request merged**